src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1304,14 +1300,21 @@ def parse(
                     # Incomplete match.
                     # For now this means the parsing has failed. Lets add the unmatched
                     # bit at the end as something unparsable.
-                    # TODO: Do something more intelligent here.
+                    # NOTE: Don't claim any additional whitespace in the failed match.
+                    _idx = 0
+                    for _idx in range(len(m.unmatched_segments)):
```

Comment:
```suggestion
                    for _idx in range(len(m.unmatched_segments)):
```

Seems like no need to set this to zero before doing the for loop.

Reply:
I've got into the habit of doing it this way in case the tuple is empty, perhaps over cautiously. I'll leave it as is for now because I might rethink this whole method anyway